### ISSUE TITLE
Recursive generic TypeVar cannot be used within the generic class context

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#2600]
**Describe the bug**

This bug is best explained with a code example:

```py
from typing import Generic, TypeVar

A = TypeVar('A')
B = TypeVar('B')

class X(Generic[A, B]):
    _dict: dict[A, B]
    _pair: 'X[B, A]'

    def method(self, a: A, b: B) -> None:
        self._pair._dict[b]  # error

x = X[int, str]()
x._pair._dict['foo']  # accepted
```

**To Reproduce**

```
example.py:11:9 - error: Argument of type "B@X" cannot be assigned to parameter "__k" of type "A@X" in function "__getitem__"
    Type "B@X" cannot be assigned to type "A@X" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.789sec
```

**Expected behavior**

No errors.

**VS Code extension or command-line**

Command-line version: 1.1.189

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.193, which I just published. It will also be included in the next release of pylance.
